src/Action/Server.hs

```diff
@@ -1,7 +1,10 @@
 {-# LANGUAGE ViewPatterns, TupleSections, RecordWildCards, ScopedTypeVariables, PatternGuards #-}
 {-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE MultiWayIf #-}
+{-# OPTIONS_GHC -Wall #-}
```

Comment:
I generally find this a bad thing to put in the source code directly - what is in Wall changes, and not all the things in there are a good idea.

Reply:
I didn’t want to fix warnings in the whole codebase for now, so I only scoped it to this module; do you want to just remove `-Wall` or have something else in mind?

Reply:
I opted for manuall disabling the shadowing warnings & pattern warnings